src/core/tiledmesh/qgstiledmeshboundingvolume.cpp

```diff
@@ -0,0 +1,252 @@
+
+  if ( transform.isValid() && !transform.isShortCircuited() )
+  {
+    ext->transform( transform, direction, true );
+  }
+  ext->dropZValue();
+  polygon->setExteriorRing( ext.release() );
+
+  return polygon.release();
+}
+
+
+//
+// QgsTiledMeshNodeBoundingVolumeBox
+//
+
+QgsTiledMeshNodeBoundingVolumeBox::QgsTiledMeshNodeBoundingVolumeBox( const QgsOrientedBox3D &box )
+  : mBox( box )
+{
+}
+
+Qgis::TiledMeshBoundingVolumeType QgsTiledMeshNodeBoundingVolumeBox::type() const
+{
+  return Qgis::TiledMeshBoundingVolumeType::OrientedBox;
+}
+
+QgsBox3d QgsTiledMeshNodeBoundingVolumeBox::bounds( const QgsCoordinateTransform &transform, Qgis::TransformDirection direction ) const
+{
+  if ( transform.isValid() && !transform.isShortCircuited() )
+  {
+    const QVector< QgsVector3D > corners = mBox.corners();
+    QVector< double > x;
+    x.reserve( 8 );
+    QVector< double > y;
+    y.reserve( 8 );
+    QVector< double > z;
+    z.reserve( 8 );
+    for ( int i = 0; i < 8; ++i )
+    {
+      x.append( corners[i].x() );
+      y.append( corners[i].y() );
+      z.append( corners[i].z() );
+    }
+    transform.transformInPlace( x, y, z, direction );
+
+    const auto minMaxX = std::minmax_element( x.constBegin(), x.constEnd() );
+    const auto minMaxY = std::minmax_element( y.constBegin(), y.constEnd() );
+    const auto minMaxZ = std::minmax_element( z.constBegin(), z.constEnd() );
+    return QgsBox3d( *minMaxX.first, *minMaxY.first, *minMaxZ.first, *minMaxX.second, *minMaxY.second, *minMaxZ.second );
+  }
+  else
+  {
+    return mBox.extent();
+  }
+}
+
+QgsTiledMeshNodeBoundingVolumeBox *QgsTiledMeshNodeBoundingVolumeBox::clone() const
+{
+  return new QgsTiledMeshNodeBoundingVolumeBox( *this );
+}
+
+QgsAbstractGeometry *QgsTiledMeshNodeBoundingVolumeBox::as2DGeometry( const QgsCoordinateTransform &transform, Qgis::TransformDirection direction ) const
+{
+  std::unique_ptr< QgsPolygon > polygon = std::make_unique< QgsPolygon >();
+
+  const QVector< QgsVector3D > corners = mBox.corners();
+  QVector< double > x;
+  x.reserve( 8 );
+  QVector< double > y;
+  y.reserve( 8 );
+  QVector< double > z;
+  z.reserve( 8 );
+  for ( int i = 0; i < 8; ++i )
+  {
+    x.append( corners[i].x() );
+    y.append( corners[i].y() );
+    z.append( corners[i].z() );
+  }
+  if ( transform.isValid() && !transform.isShortCircuited() )
+  {
+    transform.transformInPlace( x, y, z, direction );
+  }
+
+  const QVector<double> x2DCenter
+  {
+    0.5 * ( x[0] + x[4] ),
+    0.5 * ( x[1] + x[5] ),
+    0.5 * ( x[2] + x[6] ),
+    0.5 * ( x[3] + x[7] ),
+    0.5 * ( x[0] + x[4] ),
+  };
+  const QVector<double> y2DCenter
+  {
+    0.5 * ( y[0] + y[4] ),
+    0.5 * ( y[1] + y[5] ),
+    0.5 * ( y[2] + y[6] ),
+    0.5 * ( y[3] + y[7] ),
+    0.5 * ( y[0] + y[4] ),
+  };
+
+  std::unique_ptr< QgsLineString > ext = std::make_unique< QgsLineString >( x2DCenter, y2DCenter );
+  polygon->setExteriorRing( ext.release() );
+  return polygon.release();
```

Comment:
To properly represent OBB in plane, we should probably project all 8 corner points to the plane and then create a convex hull of those, rather than converting to a 2D oriented box? I am not sure how closely this approximates the true shape...

Reply:
Indeed :laughing: 

I'm stuck trying to find any real world data to test this with though -- all the datasets I can find with a OBB volume also have a transform present (which isn't handled in this code yet). Do you have any sample datasets with an OBB but not transform I can test with?

Reply:
google tiles have OBB and no transform - but from my experience the boxes (at least when zoomed in) are pretty much aligned with axes after they get converted to a planar CRS :-)

Reply:
@wonder-sk fix is in https://github.com/qgis/QGIS/pull/53863